test/simulation/xud_test.go

```diff
@@ -377,6 +377,9 @@ func launchNetwork(noBalanceChecks bool) (*xudtest.NetworkHarness, func()) {
 		log.Fatalf("cannot start xud network: %v", err)
 	}
 
+	// Wait a bit to avoid calls to xud nodes while still initializing.
+	time.Sleep(3 * time.Second)
```

Comment:
It would be good to handle this with a check & retry approach rather than a single wait. Depending on the system 3 seconds could be too much and slow down the tests, but in some cases it could be not enough and cause flake. It's probably not a big deal and doesn't have to be changed in this PR necessarily, but it'd be nice to at least have a TODO here or maybe add to the list of sim test improvements.

Reply:
I actually encountered this just once (xud returning `xud is starting` error), so I wasn't thinking of doing a more robust solution. 

I'll add it as a task to the roadmap.

Reply:
I actually encountered this just once (xud returning "xud is starting" error), so I wasn't thinking of doing a more robust solution.

I'll add it as a task to the roadmap.